botocore/httpsession.py

```diff
@@ -318,6 +330,15 @@ def send(self, request):
             manager = self._get_connection_manager(request.url, proxy_url)
             conn = manager.connection_from_url(request.url)
             self._setup_ssl_cert(conn, request.url, self._verify)
+            if ensure_boolean(
+                os.environ.get('BOTO_EXPERIMENTAL__ADD_PROXY_HOST_HEADER', '')
```

Comment:
I think the env var would likely go away at that point because "graduation" in this case is it just becomes the default way we handle CONNECT requests to proxies. Perhaps this sticks around as a vestigial way to disable the feature but I think that can remain TBD for now.